CMakeLists.txt

```diff
@@ -9,6 +9,7 @@ endif()
 
 # Determine whether CCCL is the top-level project or included into
 # another project via add_subdirectory()
+set(CCCL_TOPLEVEL_PROJECT OFF)
 if ("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_LIST_DIR}")
   set(CCCL_TOPLEVEL_PROJECT ON)
 endif()
```

Comment:
I was wondering whether we could collapse this using a generator expression, but it doesn't feel worth it even if we could

Reply:
> I was wondering whether we could collapse this using a generator expression, 

We can't use generator expressions as we need to evaluate the variable at configure time